### PR TITLE
robot: fix Python client import

### PR DESCRIPTION
**Pull request by antisaling**
This fixes Robot tests for the Python client.

## Description
This fixes the imported python client on Robot tests to match what's exposed after adding a higher-level API.

## How Has This Been Tested?
I haven't managed to run robot locally successfully yet, but it seems the import problem has been fixed.

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by antisaling** [comment]
@tarilabs I haven't been able to run this locally per the instructions you gave me:

```sh
docker run -p 9090:8080 --env METADATA_STORE_SERVER_CONFIG_FILE=/tmp/shared/conn_config.pb --volume../test/bdd:/tmp/shared gcr.io/tfx-oss-public/ml_metadata_store_server:1.14.0
```

then
```sh
make build
./model-registry proxy
```

then
```sh
TEST_MODE=Python robot test
```

Am I doing something wrong? Is there a better way to run this?
From what I've noticed this fails because the MLMD types haven't been created, if so, I might be missing some step of the setup.

**Comment by tarilabs** [comment]
> Am I doing something wrong? Is there a better way to run this?
> From what I've noticed this fails because the MLMD types haven't been created

When the proxy connects, it automatically creates the MLMD _type(s).
Do you get in the proxy the log line which shows successfull connection to MLMD gRPC?
If you connect to the SQLite db file, do you see the entries?

**Review by lampajr — APPROVED** [review]
@isinyaaa I think the issue is in your volume, you are passing `/test/bdd` but you should pass the local dir where `conn_config.pb` is stored which is `test/config/ml-metadata/`, let's try with this change and it should work.

I tried locally and `robot` tests are working properly, therefore LGTM!!

**Comment by lampajr** [comment]
> I tried locally and `robot` tests are working properly, therefore LGTM!!

Nevermind, I did not use the proper binary.. they are not working running: `TEST_MODE=python robot test/robot`

File: test/robot/ModelRegistry.py [hunk 1]
```diff
@@ -2,29 +2,30 @@
...
     def upsert_registered_model(self, registered_model) -> str:
         p = RegisteredModel(None)
         for key, value in registered_model.items():
             setattr(p, key, value)
         return super().upsert_registered_model(p)
-    
+
     def upsert_model_version(self, model_version, registered_model_id: str) -> str:
         write_to_console(model_version)
         p = ModelVersion(ModelArtifact("", ""), "", "")
```
Comment (lampajr) [inline test/robot/ModelRegistry.py]:
```suggestion
        p = ModelVersion("", "", "")
```

**Review by lampajr — CHANGES_REQUESTED** [review]
This should fix the issue for `MRandLogicalModel.robot`, still checking the other one

File: test/robot/ModelRegistry.py (hunk 1 above)
Comment (lampajr) [reply test/robot/ModelRegistry.py]: Ok I think this is enough to fix all issues, the other errors are related to unimplemented keywords in `MRkeywords` that I think can be implemented in a separated PR

Comment (antisaling) [reply test/robot/ModelRegistry.py]: thanks, Andrea :) I'm still not too familiar with the framework, but I don't find the errors very informative either way

Comment (lampajr) [reply test/robot/ModelRegistry.py]: No worries, me neither hehe

**Review by lampajr — APPROVED** [review]
LGTM, robot tests are working for me

**Review by tarilabs — APPROVED** [review]
Looks good with me.

Made sure to have installed: `pip install robotframework robotframework-requests PyYAML`
and of course all python MR client dep and related.

Tested locally with.

In one terminal
```
docker run -p 9090:8080 --env METADATA_STORE_SERVER_CONFIG_FILE=/tmp/shared/conn_config.pb --volume ./test/config/ml-metadata:/tmp/shared gcr.io/tfx-oss-public/ml_metadata_store_server:1.14.0
```

In another terminal
```
./model-registry proxy
I1213 18:35:25.649365   14668 proxy.go:31] proxy server started at localhost:8080
I1213 18:35:25.649665   14668 proxy.go:37] connecting to MLMD server localhost:9090..
I1213 18:35:25.678205   14668 proxy.go:49] connected to MLMD server
```

In a third terminal:
```
robot UserStory.robot
TEST_MODE=Python robot MRandLogicalModel.robot
robot MRandLogicalModel.robot
```

image

**Comment by antisaling** [comment]
thanks a lot for the instructions, Matteo :) I managed to run it as well

**Comment by tarilabs** [comment]
> thanks a lot for the instructions, Matteo :) I managed to run it as well

Always glad to help @isinyaaa , teamwork to the moon 🚀 🌙

**Pull request merged**